src/Microsoft.IdentityModel.Tokens/Encryption/Cng.cs

```diff
@@ -0,0 +1,189 @@
+
+            return new WindowsCryptographicException(hr, message);
+        }
+
+        private sealed class WindowsCryptographicException : CryptographicException
+        {
+            public WindowsCryptographicException(int hr, string message)
+                : base(message)
+            {
+                HResult = hr;
+            }
+
+            public WindowsCryptographicException(string message) : base(message)
+            {
+            }
+
+            public WindowsCryptographicException(string message, Exception innerException) : base(message, innerException)
+            {
+            }
+
+            public WindowsCryptographicException()
+            {
+            }
+        }
+    }
+
+    internal static class AesBCryptModes
+    {
+        internal static Lazy<SafeAlgorithmHandle> OpenAesAlgorithm(string cipherMode)
+        {
+            return new Lazy<SafeAlgorithmHandle>(() =>
+            {
+                SafeAlgorithmHandle hAlg = Cng.BCryptOpenAlgorithmProvider(Cng.BCRYPT_AES_ALGORITHM, null, Cng.OpenAlgorithmProviderFlags.NONE);
+                hAlg.SetCipherMode(cipherMode);
+
+                return hAlg;
+            });
+        }
+    }
+
+    internal static class CryptographicOperations
+    {
+        [MethodImpl(MethodImplOptions.NoInlining | MethodImplOptions.NoOptimization)]
+        public static void ZeroMemory(byte[] buffer)
+        {
+            // NoOptimize to prevent the optimizer from deciding this call is unnecessary
+            // NoInlining to prevent the inliner from forgetting that the method was no-optimize
+            Array.Clear(buffer, 0, buffer.Length);
+        }
+    }
+
+    //
+    // Interop layer around Windows CNG api.
+    //
+    internal static class Cng
+    {
+        [Flags]
+        public enum OpenAlgorithmProviderFlags : int
+        {
+            NONE = 0x00000000,
+            BCRYPT_ALG_HANDLE_HMAC_FLAG = 0x00000008,
+        }
+
+        public const string BCRYPT_AES_ALGORITHM = "AES";
+
+        public const string BCRYPT_CHAIN_MODE_GCM = "ChainingModeGCM";
+
+        public static SafeAlgorithmHandle BCryptOpenAlgorithmProvider(string pszAlgId, string pszImplementation, OpenAlgorithmProviderFlags dwFlags)
+        {
+            SafeAlgorithmHandle hAlgorithm;
+            NTSTATUS ntStatus = Interop.BCrypt.BCryptOpenAlgorithmProvider(out hAlgorithm, pszAlgId, pszImplementation, (int)dwFlags);
+            if (ntStatus != NTSTATUS.STATUS_SUCCESS)
+                throw LogHelper.LogExceptionMessage(CreateCryptographicException(ntStatus));
+            return hAlgorithm;
+        }
+
+        public static void SetCipherMode(this SafeAlgorithmHandle hAlg, string cipherMode)
+        {
+            NTSTATUS ntStatus = Interop.BCrypt.BCryptSetProperty(hAlg, BCryptPropertyStrings.BCRYPT_CHAINING_MODE, cipherMode, (cipherMode.Length + 1) * 2, 0);
+
+            if (ntStatus != NTSTATUS.STATUS_SUCCESS)
+            {
```

Comment:
Be consistent with braces, no need with single statement in clause.

Reply:
Fixed it.